### ISSUE TITLE
基于lora训练当优化器不是lion时一直loss一直是nan，感觉是哪里有问题

### ISSUE THREAD

**Issue by ericperfect** [ssbuild/chatglm_finetuning#71]
基于lora训练当优化器不是lion时一直loss一直是nan，感觉是哪里有问题，和学习率无关应该，同样的数据和学习率用别人的代码不是nan

**Comment by weizhenzhao** [comment]
<img width="1633" alt="image" src="https://user-images.githubusercontent.com/13706689/227708926-54cb7b9e-c039-452b-92f9-039c04ed8aca.png">
<img width="1526" alt="image" src="https://user-images.githubusercontent.com/13706689/227708997-d0f493e1-bc70-4936-938b-6e8051b3c778.png">

@ssbuild 大佬我也遇到了这个问题，我训练了11894条数据，用的lora 模式，28层。
`sql

train_info_args = {
    'devices': 2,
    'data_backend': 'record',
    'model_type': 'chatglm',
    # 预训练模型路径 , 从0训练，则置空
    'model_name_or_path': '/root/autodl-tmp/chat-glm/chatglm-6b',
    'config_name': './config/config_small.json',
    'tokenizer_name': '/root/autodl-tmp/chat-glm/chatglm-6b',
    'convert_onnx': False, # 转换onnx模型
    'do_train': True,
    'train_file':  [ './data/train.json'],
    'max_epochs': 10,
    'max_steps': -1,
    'optimizer': 'lion', # one of adamw,adam,lamb,lion
    'scheduler_type': 'linear',
    'optimizer_betas': (0.9, 0.999),
    'train_batch_size': 4,
    'eval_batch_size': 2,
    'test_batch_size': 2,
    'learning_rate': 2e-5,  #
    'adam_epsilon': 1e-8,
    'gradient_accumulation_steps': 1,
    'max_grad_norm': 1.0,
    'weight_decay': 0,
    'warmup_steps': 0,
    'output_dir': './output',
    'max_seq_length': 1024, # 如果资源充足，推荐长度2048 与官方保持一致
    'max_target_length': 100,  # 预测最大长度, 保留字段
    'use_fast_tokenizer': False,
    'do_lower_case': False,

    ##############  lora模块
    'with_lora': True,  # 是否启用lora模块
    'inference_mode': False, # 推理模型, 不需要手动设置
    'r': 8,
    'target_modules': ['query_key_value'],
    'target_dtype': '16',
    'lora_alpha': 32,
    # 'enable_lora': [True],
    'enable_lora': None,
    'lora_dropout': 0.1,
    'bias': 'none',  # Bias type for Lora. Can be 'none', 'all' or 'lora_only'"
}


`

看起来loss没有收敛的趋势呀。
用lora模式真的能收敛吗？

**Comment by ssbuild** [comment]
@ericperfect 

如果优化器不是lion ,可以尝试将config/config_small.json  precision 改为32 ，然后修改 train.py Trainner 添加参数 precision=16 或者precision='bf16'   try it

**Comment by ericperfect** [comment]
谢谢，我试试看

**Comment by ericperfect** [comment]
@ssbuild 改了config文件precision 为32 出现一个尴尬的问题就是一张24G的显卡加载模型就oom

**Comment by ssbuild** [comment]
> @ssbuild 改了config文件precision 为32 出现一个尴尬的问题就是一张24G的显卡加载模型就oom

那就没太好方法啦，loss nan应该是精度溢出导致。

**Comment by ericperfect** [comment]
@weizhenzhao 请问下你这个tensorboard是怎么支持的

**Comment by ssbuild** [comment]
> @weizhenzhao 请问下你这个tensorboard是怎么支持的

output 子文件夹里面

**Comment by leoluopy** [comment]
@ssbuild 
有没有什么方式可以避免精度溢出？ norm clip ？
